libpod/container_internal_linux.go

```diff
@@ -1717,11 +1717,27 @@ func (c *Container) generateCurrentUserPasswdEntry() (string, int, int, error) {
 	// If the user's actual home directory exists, or was mounted in - use
 	// that.
 	homeDir := c.WorkingDir()
-	if MountExists(c.config.Spec.Mounts, u.HomeDir) {
-		homeDir = u.HomeDir
+	hDir := u.HomeDir
+	for hDir != "/" {
+		if MountExists(c.config.Spec.Mounts, hDir) {
+			homeDir = u.HomeDir
+			break
+		}
+		hDir = filepath.Dir(hDir)
```

Comment:
me thinks
```suggestion
		homeDir = filepath.Dir(hDir)
```
hDir doesn't seem to be referred to after this line otherwise.

Reply:
If hDir is volume mounted into the container then
```
+                       homeDir = u.HomeDir
+                       break
```
Is executed which sets the homeDir to the directory from the host as opposed to the CWD.